river/Seat.zig

```diff
@@ -436,6 +441,11 @@ fn addPointer(self: Self, device: *wlr.InputDevice) void {
     self.cursor.wlr_cursor.attachInputDevice(device);
 }
 
+fn addTouch(self: *Self, device: *wlr.InputDevice) void {
+    self.cursor.wlr_cursor.attachInputDevice(device);
+    self.has_touch = true;
```

Comment:
I'm not sure how to handle the destroy signal.

Reply:
The destroy signal on `wlr.InputDevice` is the one I was talking about here.

Reply:
Should `Seat.zig` handle this or should it be handled in a `Touch.zig` file similar to `Keyboard.zig`? Also, when a device is destroyed, should the capabilities of the seat be updated? Shouldn't this also be the case for when a keyboard is destroyed? I'm not sure how or if this gets done right now.

Reply:
Hey, sorry it's taken so long to get back to you. It looks like we don't ever update the capabilities on removal for keybords/pointers currently. I think the right way to do this would be adding a `removeDevice()` function to `Seat.zig` and call that from `InputDevice.handleDestroy()` in `InputManager.zig`.

Anyhow, since it turns out we don't do this for *any* input devices currently feel free to leave out of this PR, I can implement that when I find the time to merge this if you like.